File: Insert a node at the head of a linked list/insert-to-head-linked-list.js
const SinglyLinkedListNode = class {
  constructor(nodeData) {
    this.data = nodeData;
    this.next = null;
  }
};

const SinglyLinkedList = class {
  constructor() {
    this.head = null;
    this.tail = null;
  }
};

function printSinglyLinkedList(node) {
  while (node != null) {
    console.log(String(node.data));
    node = node.next;
  }
}

function insertNodeAtHead(head, data) {
  const new_head = new SinglyLinkedListNode(data);

  if (head) {
    new_head.next = head;
  }

  return new_head;
}

function main() {
  const llist = new SinglyLinkedList();
  const list_data = [...Array(10).keys()];

  for (let data in list_data) {
    const llist_head = insertNodeAtHead(llist.head, data);
    llist.head = llist_head;
  }

  printSinglyLinkedList(llist.head);
}

main();

File: Insert a node at the head of a linked list/insert-to-head-linked-list.py
class SinglyLinkedListNode:
    def __init__(self, node_data):
        self.data = node_data
        self.next = None


class SinglyLinkedList:
    def __init__(self):
        self.head = None
        self.tail = None


def print_singly_linked_list(node):
    while node:
        print(str(node.data))
        node = node.next


def insertNodeAtHead(head, data):
    new_head = SinglyLinkedListNode(data)

    if head is not None:
        new_head.next = head

    return new_head


if __name__ == '__main__':
    llist_count = int(input())

    llist = SinglyLinkedList()

    for i in range(llist_count):
        llist_item = int(input())
        llist_head = insertNodeAtHead(llist.head, llist_item)
        llist.head = llist_head

    print_singly_linked_list(llist.head)
